src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs

```diff
@@ -26,11 +27,19 @@ public CalendarEventsController(ILogger<CalendarEventsController> logger, IMedia
     [ProducesResponseType(typeof(GetMemberResult), StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> Get(Guid calendarEventId)
+    public async Task<IActionResult> Get(Guid calendarEventId, [FromHeader(Name = "X-RequestedByUser")] Guid requestedByUserId)
     {
-        _logger.LogInformation("AAN Hub API: Received command to get calendar event by {calendarEventId}: ", calendarEventId);
+        _logger.LogInformation("AAN Hub API: Received command from User ID {requestedByUserId} to get calendar event by event ID {calendarEventId}", requestedByUserId, calendarEventId);
 
-        var response = await _mediator.Send(new GetCalendarEventByIdQuery(calendarEventId));
+        var response = await _mediator.Send(new GetCalendarEventByIdQuery(calendarEventId, requestedByUserId));
+
+        if (!response.IsValidResponse)
+        {
+            foreach (var errorMessage in response.Errors.Select(e => e.ErrorMessage))
+            {
+                _logger.LogError("AAN Hub API: {errorMessage}. Request came from User ID {requestedByuserId}", errorMessage, requestedByUserId);
```

Comment:
Here I'm logging both the error and the user ID of the request, since that would make it easier to track errors. The calendar event ID is already being logged.

Reply:
I thought those details would be in the header anyway? This seems like a lot of logging that clutters this controller.  We don't usually do this.   I'll add a note here so @DevenKShah can comment, and will defer to his decision

Reply:
I agree with Mark, there is no need to log any of these validation failures. Also these are not errors and should not be logged as errors. Please refer [WoW ](https://skillsfundingagency.atlassian.net/wiki/spaces/NDL/pages/3929210881/AAN+Dev+Team+Ways+of+working#Error) for logging tips.

Reply:
Invalid response logging removed